### PR TITLE
ENT-9083: Updated docs to include new behaviour of package[update]smatching

### PR DESCRIPTION
**Pull request by larsewi**

### CODE REVIEW

File: reference/functions/packagesmatching.markdown [hunk 1]
```diff
@@ -42,8 +42,10 @@ some desired packages, and finally reports if they are installed.
...
-At no time will both standard and legacy data be available to these functions.
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
+This change enables the usage of these policy functions in standalone policy files. But please note that you still need the default package inventory attribute specified in the policy framework for the software inventory databases to exist in the first place and for them to be maintained/updated.
+If there is no `package_inventory` attribute (such as on package module unsupported platforms) and there are no software inventory databases available in `$(sys.statedir)` then the legacy package methods data will be used instead.
+At no time will both the standard and the legacy data be available to these functions silmontaneously.
```
Comment (craigcomstock) [inline reference/functions/packagesmatching.markdown:48]:
```suggestion
At no time will both the standard and the legacy data be available to these functions simultaneously.
```

File: reference/functions/packageupdatesmatching.markdown [hunk 2]
```diff
@@ -39,8 +39,10 @@ this:
...
-At no time will both standard and legacy data be available to these functions.
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
+This change enables the usage of these policy functions in standalone policy files. But please note that you still need the default package inventory attribute specified in the policy framework for the software inventory databases to exist in the first place and for them to be maintained/updated.
+If there is no `package_inventory` attribute (such as on package module unsupported platforms) and there are no software inventory databases available in `$(sys.statedir)` then the legacy package methods data will be used instead.
+At no time will both the standard and the legacy data be available to these functions silmontaneously.
```
Comment (craigcomstock) [inline reference/functions/packageupdatesmatching.markdown:45]:
```suggestion
At no time will both the standard and the legacy data be available to these functions simultaneously.
```

**Review by craigcomstock — CHANGES_REQUESTED** [review]

File: reference/functions/packagesmatching.markdown [hunk 3]
```diff
@@ -42,8 +42,10 @@ some desired packages, and finally reports if they are installed.
 When `package_inventory` in `body common control` is configured, CFEngine will record the packages installed and the package updates available for the configured package modules.
 In the [Masterfiles Policy Framework][Masterfiles Policy Framework] `package_inventory` will be [configured](https://github.com/cfengine/masterfiles/blob/3dc1f629544b24261975ecf86e02554d4daf346e/promises.cf.in#L92) to the default for the hosts platform.
 Since only one `body common control` can be present in a policy set any bundles which use these functions will typically need to execute in the context of a full policy run.
-If there is no `package_inventory` attribute such as on package module unsupported platforms or when a policy entry file other than promises.cf is selected with the `--file -f` argument then the legacy package methods data will be used.
-At no time will both standard and legacy data be available to these functions.
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
```
Comment (nickanderson) [inline reference/functions/packagesmatching.markdown:45]: 3.22? Won't this be 3.23?

And then if it's backported, 3.18.6, 3.21.3

File: reference/functions/packageupdatesmatching.markdown [hunk 4]
```diff
@@ -39,8 +39,10 @@ this:
...
-At no time will both standard and legacy data be available to these functions.
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
+This change enables the usage of these policy functions in standalone policy files. But please note that you still need the default package inventory attribute specified in the policy framework for the software inventory databases to exist in the first place and for them to be maintained/updated.
+If there is no `package_inventory` attribute (such as on package module unsupported platforms) and there are no software inventory databases available in `$(sys.statedir)` then the legacy package methods data will be used instead.
+At no time will both the standard and the legacy data be available to these functions simultaneously.
```
Comment (nickanderson) [inline reference/functions/packageupdatesmatching.markdown:45]: Seems like this should be noted in the History section

File: reference/functions/packagesmatching.markdown (hunk 3 above)
Comment (nickanderson) [reply reference/functions/packagesmatching.markdown:45]: Also seems liek this should be mentioned in the history section.

**Review by nickanderson — APPROVED** [review]

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/376/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/376/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-376/output/_site/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/377/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/377/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-377/output/_site/

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by larsewi** [comment]
Cherry picked to
 - https://github.com/cfengine/documentation/pull/3073
 - https://github.com/cfengine/documentation/pull/3074

**Pull request merged**